app/main.py

```diff
@@ -1,3 +1,14 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    items_in_cached = {}
+
+    def wrapper(*args) -> Callable:
+        if args in items_in_cached.keys():
+            print("Getting from cache")
+            return items_in_cached[args]
+        print("Calculating new result")
+        items_in_cached.update({args: func(*args)})
```

Comment:
Update is redundant in this case. You can do:
items_in_cached[args] = func(*args)

Reply:
Also you can use else and than remove the same code: 10 and 13 lines